src/pybind/matrix/vector_pybind.cc

```diff
@@ -44,20 +45,66 @@ void pybind_vector(py::module& m) {
            [](const VectorBase<float>& v, int i) { return v(i); })
       .def("__setitem__",
            [](VectorBase<float>& v, int i, float val) { v(i) = val; })
-      .def("numpy", [](VectorBase<float>* v) {
-        return py::array_t<float>(
-            {v->Dim()},       // shape
-            {sizeof(float)},  // stride in bytes
-            v->Data(),        // ptr
-            py::none());      // pass a base object to avoid copy!
-        // (fangjun): the base object can be anything containing a non-null
-        // ptr. I cannot think of a better way than to pass a `None` object.
-        //
-        // `numpy` instead of `Numpy`, `ToNumpy` or `SomeOtherName` is used here
-        // because we want to follow the style in PyKaldi and PyTorch using the
-        // method `numpy()` to convert a matrix/tensor to a numpy array.
+      .def("numpy",
+           [](VectorBase<float>* v) {
+             return py::array_t<float>(
+                 {v->Dim()},       // shape
+                 {sizeof(float)},  // stride in bytes
+                 v->Data(),        // ptr
+                 py::none());      // pass a base object to avoid copy!
+             // (fangjun): the base object can be anything containing a non-null
+             // ptr. I cannot think of a better way than to pass a `None`
+             // object.
+             //
+             // `numpy` instead of `Numpy`, `ToNumpy` or `SomeOtherName` is used
+             // here because we want to follow the style in PyKaldi and PyTorch
+             // using  the  method `numpy()` to convert a matrix/tensor to a
+             // numpy array.
+           })
+      .def("to_dlpack", [](VectorBase<float>* v) {
```

Comment:
I think now is the best time to add documentation for this, while it's still fresh in your mind.

Reply:
do we need to create a directory`pybind/doc` and puts a `kaldi_pybind.dox` in it ?